<launch>
    <arg name="floor_name" default="floor1" doc="Default 2D map floor name" />
    <arg name="launch_rviz" default="true" doc="Controls whether to launch RViz from this file"/>
    <arg name="launch_pcd" default="true" doc="Set to 'false' to disable the 3D point cloud publisher"/>
    <!-- <arg name="json_file" default="waypoints.json" /> -->
    <arg name="pcd_file" default="room_zheng.ply" />
    <arg name="map_frame" default="map" />

    <!-- 2D nav map server-->
    <node pkg="map_server" type="map_server" name="view_map" args="$(find mixed_nav)/resources/floors/$(arg floor_name)/$(arg floor_name).yaml" />
    <node pkg="rviz" type="rviz" name="map_rviz" args="-d $(find mixed_nav)/rviz/view_map.rviz" if="$(arg launch_rviz)" />

    <!-- 3D map server-->
    <node pkg="mixed_nav" type="pcd_map_publisher.py" name="pcd_map_publisher_py" output="screen" if="$(arg launch_pcd)">
        <param name="pcd_file_path" value="$(find mixed_nav)/resources/floors/$(arg pcd_file)" />
        <param name="frame_id" value="$(arg map_frame)" />
        <param name="topic_name" value="/pointcloud_map" />
        <param name="max_points_before_downsample" value="1500000" />
        <param name="leaf_size" value="0.1" />
        <param name="initial_z_offset" value="0.0" /> <param name="leaf_size" value="0.1" />
    </node>

    <!-- path display -->
    <node pkg="mixed_nav" type="visualize_nav_paths.py" name="nav_path_visualizer" output="screen">
        <param name="initial_path_key" value="$(arg floor_name)" />
    </node>

    <!-- floor switcher -->
    <node pkg="mixed_nav" type="floor_switcher.py" name="floor_switcher_node" output="screen" />

    <!-- 2D Map mamager -->
    <node pkg="mixed_nav" type="dynamic_map_loader.py" name="dynamic_map_loader" output="screen">
        <param name="initial_map" value="$(arg floor_name)" />
    </node>

</launch>